sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py

```diff
@@ -5,37 +5,36 @@
 # --------------------------------------------------------------------------
 
 from io import BytesIO
-from typing import Union, Iterable, IO  # pylint: disable=unused-import
+from typing import Any, Dict, Generator, IO, Iterable, Optional, Union
 
-from ._shared.avro.datafile import DataFileReader
 from ._shared.avro.avro_io import DatumReader
+from ._shared.avro.datafile import DataFileReader
 
 
 class BlobQueryReader(object):  # pylint: disable=too-many-instance-attributes
-    """A streaming object to read query results.
-
-    :ivar str name:
-        The name of the blob being quered.
-    :ivar str container:
-        The name of the container where the blob is.
-    :ivar dict response_headers:
-        The response_headers of the quick query request.
-    :ivar bytes record_delimiter:
-        The delimiter used to separate lines, or records with the data. The `records`
-        method will return these lines via a generator.
-    """
+    """A streaming object to read query results."""
+
+    name: str
+    """The name of the blob being quered."""
+    container: str
+    """The name of the container where the blob is."""
+    response_headers: Dict[str, Any]
+    """The response_headers of the quick query request."""
+    record_delimiter: str
+    """The delimiter used to separate lines, or records with the data. The `records`
+    method will return these lines via a generator."""
 
     def __init__(
         self,
-        name=None,
-        container=None,
-        errors=None,
-        record_delimiter='\n',
-        encoding=None,
-        headers=None,
-        response=None,
-        error_cls=None,
-    ):
+        name: str = None,  # type: ignore [assignment]
+        container: str = None,  # type: ignore [assignment]
+        errors: Any = None,
+        record_delimiter: str = '\n',
+        encoding: Any = None,
+        headers: Dict[str, Any] = None,  # type: ignore [assignment]
+        response: Any = None,
+        error_cls = None,
```

Comment:
Check type for `errors`, `encoding` (should be str), and `error_cls` (probably callable)

Reply:
Inside of `query_blob:`
```
errors = kwargs.pop("on_error", None)
error_cls = kwargs.pop("error_cls", BlobQueryError)
encoding = kwargs.pop("encoding", None)
```
And then passed to the `BlobQueryReader` ctor.